operator/operator.go

```diff
@@ -116,36 +117,45 @@ func NewOperatorFromConfig(c optypes.NodeConfig) (*Operator, error) {
 		return nil, err
 	}
 
-	chainioConfig := clients.BuildAllConfig{
-		EthHttpUrl:                 c.EthRpcUrl,
-		EthWsUrl:                   c.EthWsUrl,
-		RegistryCoordinatorAddr:    c.AVSRegistryCoordinatorAddress,
-		OperatorStateRetrieverAddr: c.OperatorStateRetrieverAddress,
-		AvsName:                    AVS_NAME,
-		PromMetricsIpPortAddress:   c.EigenMetricsIpPortAddress,
+	reg := prometheus.NewRegistry()
+	rpcCallsCollector := rpccalls.NewCollector(c.OperatorAddress+OperatorSubsytem, reg)
+	var withOptionalMetrics safeclient.SafeEthClientOption
+	if c.EnableMetrics {
+		withOptionalMetrics = safeclient.WithInstrumentedCreateClient(rpcCallsCollector)
+	} else {
+		withOptionalMetrics = func(*safeclient.SafeEthClient) {}
 	}
-	sdkClients, err := clients.BuildAll(chainioConfig, ecdsaPrivateKey, logger)
+
+	ethHttpClient, err := safeclient.NewSafeEthClient(c.EthRpcUrl, logger, withOptionalMetrics)
 	if err != nil {
-		panic(err)
+		logger.Error("Cannot create http ethclient", "err", err)
+		return nil, err
 	}
-	reg := sdkClients.PrometheusRegistry
 
-	id := c.OperatorAddress + OperatorSubsytem
-	ethRpcClient, err := core.CreateEthClientWithCollector(id, c.EthRpcUrl, c.EnableMetrics, reg, logger)
+	ethWsClient, err := safeclient.NewSafeEthClient(c.EthWsUrl, logger, withOptionalMetrics)
 	if err != nil {
+		logger.Error("Cannot create ws ethclient", "err", err)
 		return nil, err
 	}
 
-	ethWsClient, err := core.CreateEthClientWithCollector(id, c.EthWsUrl, c.EnableMetrics, reg, logger)
+	sdkClients, err := chainio.BuildAll(
+		AVS_NAME,
+		c.AVSRegistryCoordinatorAddress,
+		c.OperatorStateRetrieverAddress,
+		ethHttpClient,
+		ethWsClient,
+		ecdsaPrivateKey,
+		logger,
+	)
 	if err != nil {
-		return nil, err
+		panic(err)
```

Comment:
Do not get confused with the diff: previously we were already panicking of we could not build `sdkClients`